api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupPNCRegister2019.java

```diff
@@ -0,0 +1,210 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ */
+@Component
+public class SetupPNCRegister2019 extends UgandaEMRDataExportManager {
+
+    @Autowired
+    private DataFactory df;
+
+    @Autowired
+    SharedDataDefintion sdd;
+
+    /**
+     * @return the uuid for the report design for exporting to Excel
+     */
+    @Override
+    public String getExcelDesignUuid() {
+        return "d6ab98f5-570d-45d6-ac57-bd143b162948";
+    }
+
+    @Override
+    public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+        List<ReportDesign> l = new ArrayList<ReportDesign>();
+        l.add(buildReportDesign(reportDefinition));
+        return l;
+    }
+
+    /**
+     * Build the report design for the specified report, this allows a user to override the report design by adding properties and other metadata to the report design
+     *
+     * @param reportDefinition
+     * @return The report design
+     */
+    @Override
+    public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "PNCRegister2019.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:8-10,dataset:PNC");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    @Override
+    public String getUuid() {
+        return "ffed8404-7f81-4c9e-a9eb-db2673397cc0";
+    }
+
+    @Override
+    public String getName() {
+        return "Integrated Postnatal Register 2019";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Contains mothers information after delivery";
+    }
+
+    @Override
+    public ReportDefinition constructReportDefinition() {
+        ReportDefinition rd = new ReportDefinition();
+
+        rd.setUuid(getUuid());
+        rd.setName(getName());
+        rd.setDescription(getDescription());
+        rd.addParameters(getParameters());
+        rd.addDataSetDefinition("PNC", Mapped.mapStraightThrough(dataSetDefinition()));
+        return rd;
+    }
+
+    @Override
+    public String getVersion() {
+        return "1.0.6";
+    }
+
+    @Override
+    public List<Parameter> getParameters() {
+        List<Parameter> l = new ArrayList<Parameter>();
+        l.add(df.getStartDateParameter());
+        l.add(df.getEndDateParameter());
+        return l;
+    }
+
+    private DataSetDefinition dataSetDefinition() {
+        PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+        dsd.setName("PNC");
+        dsd.addParameters(getParameters());
+        dsd.addRowFilter(Cohorts.genderAndHasAncEncounter(true, false, "fa6f3ff5-b784-43fb-ab35-a08ab7dbf074"), "startDate=${startDate},endDate=${endDate}");
+
+        PersonAttributeType phoneNumber = Context.getPersonService().getPersonAttributeTypeByUuid("14d4f066-15f5-102d-96e4-000c29c2a5d7");
```

Comment:
@solemabrothers see comments on Maternity form, these data definitions and converters should be defined globally and just used